idaes/core/util/model_diagnostics.py

```diff
@@ -4122,3 +4264,269 @@ def _collect_model_statistics(model):
+        input_mag = [self._get_value_for_sum_subexpression(i) for i in child_data]
+
+        # Next, create a copy of the external function with expected magnitudes as inputs
+        newfunc = node.create_node_with_local_data(input_mag)
+
+        # Evaluate new function and return the absolute value
+        return [value(newfunc)], mismatch, cancelling, const
+
+    def _perform_checks(self, node, child_data):
+        # Perform checks for problematic expressions
+        # First, need to check to see if any child data is a list
+        # This indicates a sum expression
+        mismatch = []
+        cancelling = []
+        const = True
+
+        for d in child_data:
+            # Collect all warnings about mismatched terms from child nodes
+            mismatch += d[1]
+            cancelling += d[2]
+
+            # We will check for cancelling terms here, rather than the sum itself, to handle special cases
+            # We want to look for cases where a sum term results in a value much smaller
+            # than the terms of the sum
+            sums = self._sum_combinations(d[0])
+            if any(i <= self._sum_tol * max(d[0]) for i in sums):
+                cancelling.append(str(node))
+
+            # Expression is not constant if any child is not constant
+            if not d[3]:
+                const = False
+
+        # Return any problematic terms found
+        return mismatch, cancelling, const
+
+    def _check_equality_expression(self, node, child_data):
+        # (In)equality expressions are a special case of sum expressions
+        # We can start by just calling the method to check the sum expression
+        vals, mismatch, cancelling, const = self._check_sum_expression(node, child_data)
+
+        # Next, we need to check for canceling terms.
+        # In this case, we can safely ignore expressions of the form constant = sum()
+        # We can also ignore any constraint that is already flagged as mismatched
+        if str(node) not in mismatch and not any(d[3] for d in child_data):
+            # No constant terms, check for cancellation
+            # First, collect terms from both sides
+            t = []
+            for d in child_data:
+                t += d[0]
+
+            # Then check for cancellations
+            sums = self._sum_combinations(t)
+            if any(i <= self._sum_tol * max(t) for i in sums):
+                cancelling.append(str(node))
+
+        return vals, mismatch, cancelling, const
+
+    def _sum_combinations(self, values_list):
+        sums = []
+        for i in chain.from_iterable(
+            combinations(values_list, r) for r in range(2, len(values_list) + 1)
+        ):
+            sums.append(abs(sum(i)))
+        return sums
+
+    node_type_method_map = {
+        EXPR.EqualityExpression: _check_equality_expression,
+        EXPR.InequalityExpression: _check_equality_expression,
+        EXPR.RangedExpression: _check_sum_expression,
```

Comment:
Don't you need special handling for Ranged (like Equality / Inequality)?  I would probably define a
```python
def _check_ranged(self, node, child_data):
    lhs_vals, lhs_mismatch, lhs_cancelling, lhs_const = self._check_equality(node, child_data[:2])
    rhs_vals, rhs_mismatch, rhs_cancelling, rhs_const = self._check_equality(node, child_data[1:])
    # Then merge the results and return
```

Reply:
I think I've fixed this - it would be good if you could double check my logic however.